add TextArea component

unixchad opened the pull request:
The style and function reused from the <Input> tag.

Need this tag for one of our projects. Just wanted to get the ball rolling on this one.

James300 commented:
Besides that one comment it looks good to me. Is the plan to update the docs project after?

sdalonzo commented on packages/core/src/TextArea.js:
```diff
@@ -0,0 +1,46 @@
+  margin: 0;
+
+  ::placeholder {
+    color: ${themeGet('colors.gray')};
```
this should be a semantic color

achoboter commented on packages/core/storybook/TextArea.js:
```diff
@@ -0,0 +1,53 @@
+        color="orange"
+        placeholder="Orange"
+      />
+      <TextArea mb={3} id="textarea-colors-7" color="blue" placeholder="Blue" />
```
We shouldn't document using old colors anymore, only semantic names.

achoboter commented on packages/core/test/TextArea.js:
```diff
@@ -0,0 +1,29 @@
+  })
+  test('it renders an input element with a red border with a color prop is passed', () => {
+    const json = rendererCreateWithTheme(
+      <TextArea id={id} color="red" />
```
Test colors should use semantic colors.

unixchad replied:
done

unixchad commented:
-refactored to use semantic colors
-added to docs/ but I am getting this when I run the server
![2019-10-11-100531_814x382_scrot](https://user-images.githubusercontent.com/3157593/66663828-04c30680-ec11-11e9-8915-bcf78f47d739.png)

I require some guidance.

unixchad commented:
It's a chicken and egg problem. The docs/ is referencing itself.

unixchad commented:
Reverted the docs/ section as discussed.

unixchad commented on packages/core/src/TextArea.js:
```diff
@@ -0,0 +1,51 @@
+  color: inherit;
+  background-color: transparent;
+  border-radius: ${themeGet('radius')};
+  border-width: 0px;
```
fixed
